sources/api/migration/migrator/src/main.rs

```diff
@@ -30,37 +30,41 @@ use simplelog::{Config as LogConfig, TermLogger, TerminalMode};
 use snafu::{ensure, OptionExt, ResultExt};
 use std::collections::HashSet;
 use std::env;
-use std::fs::{self, Permissions};
+use std::fs::{self, File, OpenOptions, Permissions};
 use std::os::unix::fs::{symlink, PermissionsExt};
 use std::os::unix::io::AsRawFd;
 use std::path::{Path, PathBuf};
 use std::process::{self, Command};
 
-use update_metadata::MIGRATION_FILENAME_RE;
+use update_metadata::{load_manifest, Direction};
+
 mod args;
-mod direction;
 mod error;
 
 use args::Args;
-use direction::Direction;
 use error::Result;
+use tough::ExpirationEnforcement;
+use url::Url;
+
+const RUNDIR: &str = "rundir";
+const TOUGH_DATASTORE: &str = "tough";
 
 // Returning a Result from main makes it print a Debug representation of the error, but with Snafu
 // we have nice Display representations of the error, so we wrap "main" (run) and print any error.
 // https://github.com/shepmaster/snafu/issues/110
 fn main() {
-    if let Err(e) = run() {
+    let args = Args::from_env(env::args());
+    if let Err(e) = run(&args) {
         eprintln!("{}", e);
         process::exit(1);
     }
 }
 
-fn run() -> Result<()> {
-    let args = Args::from_env(env::args());
-
+fn run(args: &Args) -> Result<()> {
     // TerminalMode::Mixed will send errors to stderr and anything less to stdout.
-    TermLogger::init(args.log_level, LogConfig::default(), TerminalMode::Mixed)
-        .context(error::Logger)?;
+    if let Err(e) = TermLogger::init(args.log_level, LogConfig::default(), TerminalMode::Mixed) {
+        info!("Term logger init returned an error: {}", e)
+    }
```

Comment:
I needed to do this since calling init more than once panics (spent a lot of time figuring out why my tests were panicking, but not when they ran one at a time). I could move the init outside of the run function and place it in `main` instead as an alternative fix. That's what I'll do for now.